slither/core/slither_core.py

```diff
@@ -265,13 +265,45 @@ def print_functions(self, d: str):
     ###################################################################################
     ###################################################################################
 
+    def has_ignore_comment(self, r: Dict) -> bool:
+        """
+        Check if the result has an ignore comment on the proceeding line, in which case, it is not valid
+        """
+        mapping_elements_with_lines = [
```

Comment:
Since the collection is only iterated upon, we should use [a generator expression instead of list comprehension](https://stackoverflow.com/questions/47789/generator-expressions-vs-list-comprehension).

Also, while counter-intuitive, it's better to use `{..}` instead of `dict(..)` for constructing dict as the `dict` is a function call and there is an opcode for the dict literal (`{}`) which ends up being slightly faster. Though, I wouldn't care about it much. A showcase of this below.

```
In [1]: def foo(): return dict(a=1,b=2)

In [2]: def bar(): return {'a': 1, 'b': 2}

In [3]: %timeit foo()
188 ns ± 6.21 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %timeit bar()
143 ns ± 0.64 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: import dis

In [6]: dis.dis(foo)
  1           0 LOAD_GLOBAL              0 (dict)
              2 LOAD_CONST               1 (1)
              4 LOAD_CONST               2 (2)
              6 LOAD_CONST               3 (('a', 'b'))
              8 CALL_FUNCTION_KW         2
             10 RETURN_VALUE

In [7]: dis.dis(bar)
  1           0 LOAD_CONST               1 (1)
              2 LOAD_CONST               2 (2)
              4 LOAD_CONST               3 (('a', 'b'))
              6 BUILD_CONST_KEY_MAP      2
              8 RETURN_VALUE
```

Reply:
Suggestions implemented.